core/src/marker/markerManager.cpp

```diff
@@ -72,6 +72,11 @@ bool MarkerManager::setBitmap(MarkerID markerID, int width, int height, const un
     texture->setData(bitmapData, size);
 
     marker->setTexture(std::move(texture));
+
+    // The geometry is unchanged, but the mesh must be rebuilt because DynamicQuadMesh contains
+    // texture batches as part of its data.
+    buildGeometry(*marker, m_zoom);
```

Comment:
Hmm I'd rather have the failure communicated from another function. To fully specify a marker, an application has to call `setStyling` and `setPoint`, `setPolyline`, or `setPolygon`, so to me it makes more sense to return an indication of failed geometry from the geometry setting functions.